lib/prepare.js

```diff
@@ -62,6 +62,15 @@ if (!Object.assign) Object.assign = require('object-assign')`
       : `export default function () {}`
   )
 
+  // 7. handle the theme index.js
+  const hasThemeIndexPath = fs.existsSync(options.themeApp)
+  await writeTemp(
+    'themeApp.js',
+    hasThemeIndexPath
+      ? `export { default } from ${JSON.stringify(options.themeApp)}`
+      : `export default function () {}`
+  )
+
   return options
```

Comment:
`themeApp ` and `enhanceApp` are the same except  `hasEnhancePath` and `hasThemeIndexPath`, which can be extracted.

Reply:
Not exactly the same, they are two proxy files to two different js files that may or may not exist. They will default to an empty function if they don't exist but are there to load in two different files `<theme>/index.js` and `.vuepress/enchanceApp.js` respectively into the main vue app.

Reply:
hmmm... you misunderstand my comment, I know WHY and HOW you do that...
I just want you to refactor your code like this:
```js
  async function writeEnhanceTemp (destName, srcPath, isEnhanceExist) {
    await writeTemp(
      destName,
      isEnhanceExist
        ? `export { default } from ${JSON.stringify(srcPath)}`
        : `export default function () {}`
    )
  }

  // 6. handle enhanceApp.js
  const enhanceAppPath = path.resolve(sourceDir, '.vuepress/enhanceApp.js')
  writeEnhanceTemp('enhanceApp.js', enhanceAppPath, fs.existsSync(enhanceAppPath))

  // 7. handle the theme index.js
  writeEnhanceTemp('themeApp.js', options.themeApp, fs.existsSync(options.themeApp))
```

Reply:
Ah, ok, done